<h1 align="center"><img src="https://user-images.githubusercontent.com/88287926/137457473-432503b6-1da7-4abf-93b1-b0f2aeae7a37.jpg" width="400" height="200"></h1>
HackBio Internship provides a platform for young Life Science Research Aspirants to collaborate and harness their postentials effectively in this genomic revolution using Bioinformatics and Computational Biology. The HackBio Internship (Thanos 2.0) 2021 is a Rigorous five week Internship focusing on Scientific Research, Programming, Bioinformatic Tools, Data Analysis, Networking and Social Activities.
 
 ## About The Team
<h2 align="center"> ğŸ‘¨â€ğŸ”¬ğŸ’» TEAM GENOMICS ONE ğŸ’»ğŸ‘©â€ğŸ”¬ </h2>
<p align="center">
 <br>
 <img align="center" src=""  width="400px">
<br>
	<br>
	</p>
	

## Stage 2 Task ğŸ§ğŸ§
- Reproduce a bioinformatics tutorial that correlates with our biostack (Genomics) 
- Create a comprehensive markdown of the steps followed 
- Prepare a Project Proposal Obliging the Teams Biostack

<br>

## Project Title: "Exome Sequencing Data Analysis for diagnosing a genetic disease" :computer:
Have a look at the ***New state-of-the-art molecular diagnostic genetic test-"Exome Sequencing"***

There are around 180,000 exons in humans, with a total length of approximately 30 million base pairs (30 Mb). Thus, while accounting for only 1% of the human genome, the exome is thought to include up to 85% of all disease-causing mutations. **Exome sequencing**, as an alternative to whole-genome sequencing in the detection of genetic disease, is less expensive while yet covering significantly more potential disease-causing variant sites than genotyping arrays. This is especially important in the case of rare genetic conditions, since the causal variations may present in the human population at too low a frequency to be included on genotyping arrays.

Our Team Genomics One will investigate :detective: exome sequencing data from a family triple:family_man_woman_boy: in which the boy child has a rare genetic disorder but both parents, who are consanguineous, are unaffected. Our labâ€™s goal is to figure out which genetic variant is causing the condition.

<br>

## Workflow âœï¸

### Step 1: Data preparation
Retrieve sequenced reads of father, mother and proband  in fastq format from [Zendo](https://zenodo.org/record/3054169) and import the datasets into the https://usegalaxy.org/ or https://usegalaxy.eu/

```
- Check if the datatypes were assigned correctly in the format, fastqsanger.gz 
- Rename the datasets and add tags (#father, #mother, #proband) to six of the datasets accordingly. 
- Obtain the â€˜hg19â€™ version of the human chromosome 8 sequence as the reference genome.
- Ensure the data types are specified in fasta format.

```

### Step 2: Quality control
In the FastQ format each read, representing a fragment of the library, is encoded by 4 lines:
Description
Line 1 ->  Always begins with @ followed by the information about the read

Line 2 -> The actual nucleic sequence

Line 3 -> Always begins with a + and contains sometimes the same info in line 1

Line 4 -> Has a string of characters which represent the quality scores associated with each base of the nucleic sequence; must have the same number of characters as line 2

- Run the **FastQC tool** on six of the fastq datasets with the following parameter:
```
    - â€œShort read data from your current historyâ€: all 6 FASTQ datasets selected with Multiple datasets
    - Then Execute
```
- Use the **MultiQC tool** by setting the following parameter to aggregate the raw FastQC data of all input datasets into one report:

```
     - In â€œResultsâ€
     - â€œWhich tool was used generate logs?â€: FastQC
     - In â€œFastQC outputâ€
     - â€œType of FastQC output?â€: Raw data
     - â€œFastQC outputâ€: all six RawData outputs of FastQC 
     - Then Execute
     - Inspect the webpage output produced by the tool to check if trimming/filtering is necessary before mapping the reads.

```
The quality score for each sequence is a string of characters, one for each base of the nucleic sequence, used to characterize the probability of mis-identification of each base. The score is encoded using the ASCII character table.
According to our quality check no further trimming or filtering is needed.
-All samples show a non-normal GC content distribution as it is a characteristic feature of many exome capture methods.


### Step 3: Adapter Trimming
- Use **Trimmomatic tool** on the all the fastq datasets to trim adapters and keep the settings By-default and Execute


### Step 4: Read Mapping
- Use the Map with **BWA-MEM tool** to map the reads from the â€˜Fatherâ€™, â€˜Motherâ€™, and the â€˜Probandâ€™ samples to the reference genome, respectively.
- Set the following parameters:
```
      . â€œWill you select a reference genome from your history or use a built-in index?â€: Use a built-in genome index
      . â€œUsing reference genomeâ€: Human:hg19
      . â€œSingle or Paired-end readsâ€: Paired
      . â€œSelect first set of readsâ€: the forward reads (R1) dataset of the father sample
      . â€œSelect second set of readsâ€: the reverse reads (R2) dataset of the father sample
      . â€œSet read groups information?â€: Set read groups (SAM/BAM specification)
      . â€œAuto-assignâ€: No
      . â€œRead group identifier (ID)â€: 000
      . â€œAuto-assignâ€: No
      . â€œRead group sample name (SM)â€: father
      . Then Execute
```   

- Perform the read mapping for â€˜Motherâ€™ and â€˜Probandâ€™ samples with the same parameters mentioned in the previous step with the following changes:

```     
       Mother Sample:
      . â€œRead group identifier (ID)â€: 001
      . â€œRead group sample name (SM)â€: mother
      
       Proband Sample:
      . â€œRead group identifier (ID)â€: 002
      . â€œRead group sample name (SM)â€: proband
      
```
      
### Step 5: Mapped Reads Post-processing

#### â¨€ Filtering Mapped reads
Filter the mapped reads by selecting the tool, **Filter SAM or BAM, output SAM or BAM**, and set the following parameters:

```
  - â€œSAM or BAM file to filterâ€: all 3 mapped reads datasets of the family trio, outputs of Map with BWA-MEM tool
  - â€œFilter on bitwise flagâ€: yes
  - â€œOnly output alignments with all of these flag bits setâ€: Do not select anything here!
  - â€œSkip alignments with any of these flag bits setâ€:
         âœ… â€œThe read is unmappedâ€
         âœ… â€œThe mate is unmappedâ€
  - Then Execute
  - Ensure if three new datasets are produced with one for each of the samples.

```
#### â¨€ Removing duplicate reads
Select **RmDup tool** and set the following parameters:
```
   - â€œBAM fileâ€: all 3 filtered reads datasets; the outputs of Filter SAM or BAM
   - â€œIs this paired-end or single end dataâ€: BAM is paired end
   - â€œTreat as single-endâ€: No
   - Then Execute
```
Ensure if three more new datasets are produced after this step.


### Step 6: Variant Calling 
- Select the **FreeBayes tool** and set the parameters:
```
     - â€œChoose the source for the reference genomeâ€:
     - â€œRun in batch mode?â€: Merge output VCFs
     - â€œBAM dataset(s)â€:
     - â€œUsing reference genomeâ€: Human: hg19
     - â€œLimit variant calling to a set of regions?â€: Do not limit
     - â€œChoose parameter selection levelâ€: 1. Simple diploid calling
     - Then Execute
```
Inspect the VCF outputs produced by FreeBayes.

### Step 7: FreeBayes Post-processing
- Use the **bcftools norm tool** and set the following parameters:
 ```
     - â€œVCF/BCF Dataâ€: the VCF output of FreeBayes tool
     - â€œChoose the source for the reference genomeâ€: Use a built-in genome
     - â€œReference genomeâ€: Human: hg19
     - â€œWhen any REF allele does not match the reference genome baseâ€: Ignore the problem (-w)
     - â€œLeft-align and normalize indels?â€: Yes
     - â€œPerform deduplication for the following types of variant recordsâ€: do not deduplicate any records.
     - â€œ~multiallelicsâ€: split multiallelic sites into biallelic records (-)
     - â€œsplit the following variant typesâ€: both
     - â€œoutput_typeâ€: uncompressed VCF
     - Then Execute
```
Look out for the output listing the total number of variant lines processed, along with the number of splits, realigned, and skipped records

### Step 8: Variant annotation
- Use the **SnpEff Download** to download genome annotation database  hg19
- Create a PED-formatted pedigree dataset describing the single-family sample trio in the following format:
```
      #family_id    name     paternal_id    maternal_id    sex    phenotype
        FAM         father    0              0              1      1
        FAM         mother    0              0              2      1
        FAM         proband   father         mother         1      2
```
- Use the **SnpEff eff tool** and set the following parameters:
```
    - â€œSequence changes (SNPs, MNPs, InDels)â€: the output of bcftools norm tool
    - â€œInput formatâ€: VCF
    - â€œOutput formatâ€: VCF (only if input is VCF)
    - â€œGenome sourceâ€: Locally installed reference genome
    - â€œGenomeâ€: Homo sapiens: hg19 (or a similarly named option)
    - â€œProduce Summary Statsâ€: Yes
    - Then Execute
```
- Use the **SnpSift Variant type tool** and select output of SnpEff and execute.

### Step 9: Generate GEMINI Database
- Use the **GEMINI load tool** and set the following parameters:

```
    - â€œVCF dataset to be loaded in the GEMINI databaseâ€: the output of SnpEff eff tool
    - â€œThe variants in this input areâ€: annotated with snpEff
    - â€œThis input comes with genotype calls for its samplesâ€: Yes
    
   Sample genotypes were called by Freebayes for us.
    - â€œChoose a gemini annotation sourceâ€: select the latest available annotations snapshot (most likely, there will be only one)
    - â€œSample and family information in PED formatâ€: the pedigree file prepared above
    - â€œLoad the following optional content into the databaseâ€
        âœ… â€œGERP scoresâ€
        âœ… â€œCADD scoresâ€
        âœ…â€œGene tablesâ€
        âœ…â€œSample genotypesâ€
        âœ…â€œvariant INFO fieldâ€
    
   Leave unchecked the following:
     - â€œGenotype likelihoods (sample PLs)â€
     - â€œonly variants that passed all filtersâ€
     - Then Execute
```

### Step 10: Candidate Variant Detection
- Use the **GEMINI inheritance pattern tool** and set the following parameters:
```
- â€œGEMINI databaseâ€: the GEMINI database of annotated variants; output of GEMINI load tool
- â€œYour assumption about the inheritance pattern of the phenotype of interestâ€: Autosomal recessive
        >â€œAdditional constraints on variantsâ€
        >â€œAdditional constraints expressed in SQL syntaxâ€: impact_severity != 'LOW'
        >â€œInclude hits with less convincing inheritance patternsâ€: No
        >â€œReport candidates shared by unaffected samplesâ€: No
- â€œFamily-wise criteria for variant selectionâ€: keep default settings
- In â€œOutput - included informationâ€
    .â€œSet of columns to include in the variant report tableâ€: Custom (report user-specified columns)
    .â€œChoose columns to include in the reportâ€:
        >â€œalternative allele frequency (max_aaf_all)â€
    .â€œAdditional columns (comma-separated)â€: chrom, start, ref, alt, impact, gene, clinvar_sig, clinvar_disease_name, clinvar_gene_phenotype, rs_ids
- Then Execute
```


<p align="center">
   <br>
<img align="center" alt="gif" src="" width="650">
 <br>
<br>
</p>



## References :scroll:
Wolfgang Maier, BÃ©rÃ©nice Batut, Torsten Houwaart, Anika Erxleben, BjÃ¶rn GrÃ¼ning, 2021 Exome sequencing data analysis for diagnosing a genetic disease (Galaxy Training Materials). https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html 

Batut et al., 2018 Community-Driven Data Analysis Training for Biology Cell Systems https://doi.org/10.1016%2Fj.cels.2018.05.012



## Contributors ğŸ¤
Get to know our awesome team members and their contributions ğŸ‘©â€ğŸ’»ğŸ‘¨â€ğŸ’»

| Team members	| Picture | @Slack username  | Contributions |
| ------- | --- | --- | --- |

| <h3 align="center"> Sooraj Shivakumar  </h3> | <img src="https://user-images.githubusercontent.com/88287926/137459720-8031c4ef-9eb7-4484-8892-214049ab76d2.jpg"  width="200" height="250"> | @Sooraj  |  https://in.linkedin.com/in/sooraj-s-71756510a  | 
| ------- | --- | --- | --- |
| <h3 align="center"> Negar Khalili  </h3> | <img src="https://i.ibb.co/kM4g1XH/IMG-2650.jpg"  width="200" height="250"> | @negkhalili |   https://www.linkedin.com/in/negarkhalili/  |
| ------- | --- | --- | --- |
| <h3 align="center"> Esther Opone </h3> | <img src="https://user-images.githubusercontent.com/92184734/137252429-a2c2e69b-f7aa-469f-b191-3a2e6aa33506.jpeg"  width="200" height="250"> | @thevalueadder |   https://www.linkedin.com/in/estheropone/  |
| ------- | --- | --- | --- |
| <h3 align="center"> Margaret Adedayo Opeoluwa  </h3> | <img src="https://media-exp1.licdn.com/dms/image/C4D03AQHQnOKKvGlBow/profile-displayphoto-shrink_800_800/0/1635528503200?e=1640822400&v=beta&t=qlyR_kuznZK73yC0Yb6dRIlWINy4AnvlqOF9wOTCyMQ"  width="200" height="250"/> | @LadyMarg |   https://www.linkedin.com/in/margaret-adedayo-adeogun-b6a9117a/ |


## Thank you for your Time and Patience :relaxed: 
 <p align="center">
   <br>
    <br>
<img align="center" alt="gif" src="https://github.com/arsentieva/arsentieva/blob/main/code.gif?raw=true" width="500" height="320" />
   <br>
</p>

